packages/overlay/src/integrations/sentry/components/performance/TransactionDetail.tsx

```diff
@@ -0,0 +1,159 @@
+import { useMemo, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { ReactComponent as Sort } from '~/assets/sort.svg';
+import { ReactComponent as SortDown } from '~/assets/sortDown.svg';
+import classNames from '~/lib/classNames';
+import Breadcrumbs from '~/ui/Breadcrumbs';
+import { TRANSACTION_SUMMARY_SORT_KEYS, TRANSACTION_SUMMARY_TABLE_HEADERS } from '../../constants';
+import { useSentryEvents } from '../../data/useSentryEvents';
+import { useSentryHelpers } from '../../data/useSentryHelpers';
+import { SentryTransactionEvent } from '../../types';
+
+export default function TransactionDetail({ showAll }: { showAll: boolean }) {
+  const { name } = useParams();
+
+  const events = useSentryEvents();
+  const helpers = useSentryHelpers();
+
+  const [sort, setSort] = useState({
+    active: TRANSACTION_SUMMARY_SORT_KEYS.transaction,
+    asc: false,
+  });
+
+  const toggleSortOrder = (type: string) =>
+    setSort(prev =>
+      prev.active === type
+        ? {
+            active: type,
+            asc: !prev.asc,
+          }
+        : {
+            active: type,
+            asc: false,
+          },
+    );
+
+  type TransactionsInfoComparator = (a: SentryTransactionEvent, b: SentryTransactionEvent) => number;
+  type TransactionsSortTypes = (typeof TRANSACTION_SUMMARY_SORT_KEYS)[keyof typeof TRANSACTION_SUMMARY_SORT_KEYS];
+
+  const transactionsList: SentryTransactionEvent[] = useMemo(() => {
+    if (!name) {
+      return [];
+    }
+    const COMPARATORS: Record<TransactionsSortTypes, TransactionsInfoComparator> = {
+      [TRANSACTION_SUMMARY_SORT_KEYS.transaction]: (a, b) => {
+        if (a.event_id! < b.event_id!) return -1;
+        if (a.event_id! > b.event_id!) return 1;
+        return 0;
+      },
+      [TRANSACTION_SUMMARY_SORT_KEYS.traces]: (a, b) => {
+        if (a?.contexts?.trace?.trace_id < b?.contexts?.trace?.trace_id) return -1;
+        if (a?.contexts?.trace?.trace_id > b?.contexts?.trace?.trace_id) return 1;
+        return 0;
+      },
```

Comment:
I feel like we can get rid of the non-null assertions and the optional chains if we just typed these comparator functions properly. What do you think? Non-null is not a big deal but the optional chaining definitely takes a toll on performance.

Finally (this is a side node), do you think sorting by random trace ids is actually useful? I'd much rather we have sorting via tracing timestamps which is something we keep getting asked.

Reply:
Removed these. Because as you mentioned, its not of much value to sort IDs